bin/pidtree.js

```diff
@@ -3,19 +3,133 @@
 'use strict';
 
 var os = require('os');
+var pidtree = require('..');
 
-//
-// Change the default parent PID if running
-// under Windows.
-//
-let ppid = 1;
-if (process.platform === 'win32') {
-  ppid = 0;
+function help() {
+  var help = `  Usage
+    $ pidtree <ppid>
+
+  Options
+    --list                     To print the pids as a list.
+
+  Examples
+    $ pidtree
+    $ pidtree --list
+    $ pidtree 1
+    $ pidtree 1 --list
+`;
+  console.log(help);
+}
+
+function list(ppid) {
+  pidtree(ppid === undefined ? -1 : ppid, function(err, list) {
+    if (err) {
+      console.error(err.message);
+      return;
+    }
+    console.log(list.map(e => e.pid).join(os.EOL));
+  });
 }
 
-require('..')(process.argv[2] || ppid, (err, data) => {
-  if (err) {
-    return console.error(err);
+function tree(ppid) {
+  pidtree(ppid, function(err, list) {
+    if (err) {
+      console.error(err.message);
+      return;
+    }
+
+    var fathers = {}; // HasMap of fathers
+    var tree = {}; // Adiacency HasMap
+    while (list.length > 0) {
+      var e = list.pop();
+      if (tree[e.ppid]) {
+        tree[e.ppid].push(e.pid);
+      } else {
+        tree[e.ppid] = [e.pid];
+      }
+      fathers[e.pid] = e.ppid;
+    }
+    var roots = [ppid];
+    if (ppid === -1) {
+      // Get all the roots
+      roots = Object.keys(tree).filter(node => fathers[node] === undefined);
+    }
+
+    roots.forEach(root => print(visit(tree, root)));
+  });
+
+  function visit(tree, start) {
+    // Build the tree
+    var root = {pid: start};
+    var stack = [root];
+    while (stack.length > 0) {
+      var cur = stack.pop();
+      if (!tree[cur.pid]) continue;
+      var len = tree[cur.pid].length;
+      for (var j = 0; j < len; j++) {
+        var node = {pid: tree[cur.pid][j]};
+        if (cur.children === undefined) {
+          cur.children = [node];
+        } else {
+          cur.children.push(node);
+        }
+        stack.push(node);
+      }
+      delete tree[cur.pid];
+    }
+    return root;
```

Comment:
Lots of loops no? I don't have much time to study this but looks like there's something simpler to be done. For example by already printing above (not sure)?

Reply:
This is just an iterative version of the DFS algorithm that I use to create the tree. Quite standard.
Obviously I can print the tree directly without actually creating the tree but I was tired yesterday.

Reply:
> but I was tired yesterday.

I think it reflected in your code :p.